addons/binding/org.openhab.binding.freemobilesms/src/main/java/org/openhab/binding/freemobilesms/internal/FreeMobileSmsHandler.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.freemobilesms.internal;
+
+import static org.openhab.binding.freemobilesms.internal.FreeMobileSmsBindingConstants.*;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.freemobilesms.internal.FreeMobileSmsConfiguration;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import fr.free.smsapi.Account;
+import fr.free.smsapi.Sender;
+import fr.free.smsapi.RawAccount;
+
+/**
+ * The {@link FreeMobileSmsHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Guilhem Bonnefille <guilhem.bonnefille@gmail.com> - Initial contribution
+ */
+@NonNullByDefault
+public class FreeMobileSmsHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FreeMobileSmsHandler.class);
+
+    private FreeMobileSmsConfiguration config;
+
+    private final Sender sender = new Sender();
+
+    private Account account;
```

Comment:
@NonNullByDefault and no assignment here -> does this compile? Usually you need at least a default constructed object everywhere or a @Nullable annotation. See also `config`

Reply:
Yes, it did compile, until I changed to bnd.

I have to admit I'm not really familiar to these annotations. And now, I have an issue because config is Nullable but the Sender.send expect a NonNull. I refixed with a `if (config != null)` but it doesn't change anything.

Do you have a tips to share in such situation?

Reply:
You can annotate any method or field with `@NonNullByDefault({})` to switch null checking off. That is required for example for interfaces that do not use null checking annotations.

If it's your own code you should instead annotate it with `@NonNullByDefault` (that usually leads to a chain of changed files until the entire project is annotated, but that's kind of the purpose here as well )

Reply:
Big thanks.